src/components/VideoPlayer/BaseVideoPlayer.tsx

```diff
@@ -181,6 +191,19 @@ function BaseVideoPlayer({
         currentVideoPlayerRef.current = videoPlayerRef.current;
     }, [url, currentVideoPlayerRef, isUploading]);
 
+    const isCurrentlyURLSetRef = useRef<boolean>();
+    isCurrentlyURLSetRef.current = isCurrentlyURLSet;
+
+    useEffect(
+        () => () => {
+            if (!isCurrentlyURLSetRef.current) {
+                return;
+            }
+
+            setCurrentlyPlayingURL(null);
```

Comment:
This change is causing a partial issue [here](https://github.com/Expensify/App/issues/40735), we should also return early and don't reset the `currentlyPlayingURL`  when `shouldUseSharedVideoElement` is true.